jenkins-integration/jenkins-jobs/common/scripts/background/curl-server-metrics-loop.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+
+set -x
+
+while true ; do
+  curl -sS -w "\n" -k https://localhost:8140/status/v1/services?level=debug >> /var/log/puppetlabs/puppetserver/metrics.json
```

Comment:
Would it help to ensure that the output from a full job run could be "well-formed" json?  Like, could we just write the initial `metrics.json` file with a hash containing a key whose value is an array and append the output from each curl query as an element in the array (delimited with commas).  I guess closing it off when the run is over would be a little more challenging.  Could provide an extra job-level hook for running a custom cleanup script after the stop has been completed.  Or could just rewrite this script in a such a way where it could catch the default TERM signal from the kill which is done by the stop and close off the metrics.json file before exiting out.  Actually, that latter approach might be better anyway so we don't run the risk of getting a corrupted, half-written file if the stop is stopped in the middle of writing the output from a curl request.  Suppose that wouldn't be too likely given the frequency at which the curl requests would occur.

I guess if you're going to have something downstream which knows to interpret the non-well formed format that none of this would matter much.  Would just be convenient for manual inspection of the raw file that gets archived at the end of a run, I guess.  😄